### ISSUE TITLE
Applicability in the Maritime Domain (Open Waters or Docks)

### ISSUE THREAD

**Issue by DiTo97** [wzzheng/TPVFormer#11]
Hi @wzzheng,

I'm very interested in your work, as I think it paves the way for more open-source camera-only experimentation in 3D occupancy mapping.

I have two questions for you:
1. In the comparison table with Tesla occupancy network, you reported a large discrepancy in terms of inference speed (10 ms vs. 290 ms), which would make the model not suitable for downstream real-time tasks. Why does the forward pass take so long? Maybe it's just due to the hardware and there is no actual noticeable discrepancy in place.
2. Could the methodology be applicable outside the urban domain, where the appearance of the outer environment could have drastically different properties (besides updating the semantic labels for the 3D occupancy)? In particular, we are experimenting autonomous navigation for unmmaned surface vehicles (USVs), which should be able to handle both docks scenarios as well as open waters. Thus far, we have always been quite limited in the complexity of the models, as the dataset that we have has roughly the same cardinality as yours, which was regarded as too small. Focusing on the maritime domain, if docks is not too dissimilar from an urban scenario in terms of point cloud density, open waters is orders of magnitude sparser, while retaining few objects that could provide meaningful training signal, besides the water surface, albeit being simpler to model. Would that cope well with the TPVformer model, or are there any concerns in the implementation you think would need to be addressed?

**Comment by wzzheng** [comment]
Thanks for your interest!
1. Here is a detailed inference speed of TPVFormer for your reference. The main latency lies in the TPV construction stage.  However, the current code is not optimized for speed. There is still much room for improvement. As for the comparison with Tesla, we think the main discrepancy lies in the hardware design. Their on-car hardware is perhaps designed for this, resulting in the much faster speed.
<img width="1048" alt="image" src="https://user-images.githubusercontent.com/44290343/220860746-ea23edf9-d285-4867-ab0c-4e10a3b87d36.png">
2. I'm still not quite certain about your scenario, but based on my current understanding, I do not think the point cloud density would be a large problem, as TPVFormer only relies on camera inputs in the test phase. However, the learned model might overfit the training point cloud density and output different occupancies for dock and open water scenarios.
I don't know if I'm getting this right, but I'm open to more discussions!

**Comment by DiTo97** [comment]
Thank you @wzzheng for the detailed reply,

I will investigate a bit more the code behind the construction of the TPV representation. I think I might have to try and optimize the code a bit, but not a lot, as we still have a pretty powerful computer on board.

As for the second question, my concerns are more oriented towards over fitting in training. I read the paper more carefully, and if I understood correctly you use TPVformer to generate dense semantically-rich 3D representations (varying depending on the task) supervised only on the sparse signal provided by the point clouds. The key insight is how sparse the training signal can become to train the model successfully. I am a bit worried considering the very sparse nature of open waters, and the non-homogeneity of the data between the scenarios, but I guess we will just have to train the model on our data and see if and what it's able to learn.

I will keep you posted.

**Comment by wzzheng** [comment]
> Thank you @wzzheng for the detailed reply,
> 
> I will investigate a bit more the code behind the TPV construction. I think I might have to try and optimize the code a bit, but not by much, as we still have a pretty powerful computer on board.
> 
> As for the second question, my concerns are more oriented towards over fitting in training. I read the paper more carefully, and if I understood correctly you use TPVformer to generate dense semantically-rich 3D representations (varying depending on the task) supervised only on the sparse signal provided by the point clouds. How sparse can the training signal become to still train the model successfully? I am a bit worried considering the very sparse nature of open waters, and the non-homogeneity if the data between open waters and docks scenarios, but I guess we will just have to train the model on our data and see if and what it's able to learn.
> 
> I will keep you posted.

Thanks for sharing! I look forward to your results!
